apps/src/applab/reducers.js

```diff
@@ -83,25 +83,29 @@ function instructions(state, action) {
   // TODO - we'll want to think about how to handle state that is common across
   // apps. For example, this (and eventually all of instructions) belongs in
   // a studioApps related store.
-  if (action.type === ActionType.SET_INSTRUCTIONS_IN_TOP_PANE) {
+  if (action.type === ActionType.SET_INSTRUCTIONS_IN_TOP_PANE &&
+      action.inTopPane !== state.inTopPane) {
     return _.assign({}, state, {
       inTopPane: action.inTopPane
     });
   }
 
-  if (action.type === ActionType.TOGGLE_INSTRUCTIONS_COLLAPSED) {
+  if (action.type === ActionType.TOGGLE_INSTRUCTIONS_COLLAPSED &&
+      action.collapsed !== state.collapsed) {
```

Comment:
Coincidentally it will always work since `action.collapsed` should be undefined?